File: webapi/Models/PersonalDetail.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using webapi.Framework.BaseEnums;

namespace webapi.Models
{
    [Index(nameof(Email), IsUnique = true)]
    public class PersonalDetail : CrudModel
    {
        /// <summary>
        /// Obtém ou define o primeiro nome do utilizador.
        /// </summary>
        [MaxLength(255)]
        public string? FirstName { get; set; }

        /// <summary>
        /// Obtém ou define o sobrenome do utilizador.
        /// </summary>
        [MaxLength(255)]
        public string? LastName { get; set; }

        /// <summary>
        /// Obtém ou define o e-mail do utilizador.
        /// </summary>
        [MaxLength(255), EmailAddress]
        public string? Email { get; set; }

        [Phone] 
        public string? Phone { get; set; }

        public PersonalDetail() : base()
        {
        }
    }
}

File: vueapp/src/models/personalDetail.js
import CrudModel from './crudModel.js';

class PersonalDetail extends CrudModel {
    constructor(json = {}) {
        super(json);
        this.firstName = json.firstName || null;
        this.lastName = json.lastName || null;
        this.email = json.email || null;
        this.phone = json.phone || null;
    }

    toFormData() {
        const formData = super.toFormData();
        formData.append('firstName', this.firstName);
        formData.append('lastName', this.lastName);
        formData.append('email', this.email);
        formData.append('phone', this.phone);
        return formData;
    }

    toJSON()
    {
        var data = super.toJSON()

        return {
            ...data,
            firstName: this.firstName,
            lastName: this.lastName,
            email: this.email,
            phone: this.phone,
        }
    }
}

export default PersonalDetail;